### PR TITLE
Feature/graceful shutdown

### PR DESCRIPTION
**Pull request by gedge**
adding context to web client calls,
handle signals and errors from libs
shutdown gracefully

### CODE REVIEW

File: mongo/mongo.go [hunk 1]
```diff
@@ -134,3 +138,24 @@ func (m *Mongo) UpdateJobState(id, newState string) (err error) {
 	_, err = s.DB(m.Database).C(m.Collection).Upsert(bson.M{"id": id}, update)
 	return
 }
+
+func (m *Mongo) Close(ctx context.Context) error {
```
Comment (Matt-Guest) [inline mongo/mongo.go]: Is this function needed anymore? Should it be replaced with the go-ns function.

File: cmd/dp-import-api/main.go [hunk 2]
```diff
@@ -54,23 +57,75 @@ func main() {
...
-	log.Debug("listening...", log.Data{
-		"bind_address": config.BindAddr,
-	})
+	router.Path("/healthcheck").HandlerFunc(healthcheck.Handler)
+
+	httpServer := server.New(config.BindAddr, router)
+	httpServer.HandleOSSignals = false
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, os.Kill, syscall.SIGTERM)
```
Comment (Matt-Guest) [inline cmd/dp-import-api/main.go]: Do we need to listen to `os.Kill`? As I thought that it can not be caught by a process?

Comment (gedge) [reply cmd/dp-import-api/main.go]: correct - well spotted - removed

File: mongo/mongo.go (hunk 1 above)
Comment (gedge) [reply mongo/mongo.go]: now replaced with go-ns/mongo

**Pull request merged**